src/univalue/lib/univalue.cpp

```diff
@@ -217,8 +217,7 @@ const UniValue& UniValue::operator[](size_t index) const
 void UniValue::checkType(const VType& expected) const
 {
     if (typ != expected) {
-        throw std::runtime_error{"JSON value of type " + std::string{uvTypeName(typ)} + " is not of expected type " +
-                                 std::string{uvTypeName(expected)}};
+        throw std::runtime_error{"Expected type " + std::string{uvTypeName(expected)} +  ", got " + std::string{uvTypeName(typ)}};
```

Comment:
ACK, previously this was `"JSON value is not an {} as expected"`. Then, changed in fae5ce8795080018875227aee8613677f92e99ce.

I think I wanted to use your wording, so mind to split this up into a separate pull?

Reply:
will just drop the revert (rationale: https://github.com/bitcoin/bitcoin/pull/25737#issuecomment-1200218274).

Reply:
Sure, no strong opinion. Just to clarify, you haven't "dropped the revert": It is not a revert, but a change to the string. Also, the string is still changed in this pull request.

Reply:
> Just to clarify, you haven't "dropped the revert": It is not a revert, but a change to the string. 

yeah ok. I thought that the string before fae5ce87 was the "Expected type {expected], got {type}" thus why was thinking on a revert but seems that not. We actually have two different errors for the same problematic.
Depends on the RPC command, we either throw a "JSON value of type {type} is not of expected type {expected}" or the "Expected type {expected], got {type}".

> Also, the string is still changed in this pull request.

Yeah, that is intended.

Unless someone has a strong opinion about it, will actually move the sources to only use the "JSON value of type {type} is not of expected type {expected}" error description everywhere and drop the other, less informative, one.

I think that It doesn't make sense to build any client based on a json parsing error (if the command is expecting an int and you provide a string, the command will fail anyway..), so no one should be affected by this change.